File: source/logger.cpp
#include "logger.hpp"

#include <iostream>

// Error message
void Logger::error(std::string message)
{
    had_error = true;

    std::cout << '['         << path;
    std::cout << "] Error: " << message << std::endl;
}

// Error at token
void Logger::error(Token &token, std::string message)
{
    had_error = true;

    std::cout << '['        << path << ':';
    std::cout << token.line << ':'  << token.column;
    std::cout << "] Error";

    if (token.kind == TokenKind::End)
        std::cout << " at end: ";
    else if (token.kind == TokenKind::Error)
        std::cout << ": ";
    else
        std::cout << " at '" << token.to_string() << "': ";

    std::cout << message << std::endl;
}

File: source/Logger.cs
namespace Suteki;

using System;

class Logger
{
    public string Path;

    // Show error
    public void Error(string message)
    {
        Config.HadError = true;
        Console.Error.WriteLine($"[{Path}] Error: {message}");
    }

    // Show error at token
    public void Error(Token token, string message)
    {
        Config.HadError = true;
        Console.Error.Write($"[{Path}:{token.Line}:{token.Column}] Error");

        switch (token.Kind)
        {
            case TokenKind.End:
            {
                Console.Error.Write(" at end: ");
                break;
            }

            case TokenKind.Error:
            {
                Console.Error.Write(": ");
                break;
            }

            default:
            {
                Console.Error.Write($" at '{token.Content}': ");
                break;
            }
        }

        Console.Error.WriteLine(message);
    }
}